src/App.js

```diff
@@ -1,16 +1,54 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [chatMessagesData, setChatMessagesData] = useState(chatMessages);
+  const users = [];
+  let likesCount = 0;
+  let user1 = '';
+  let user2 = '';
+
+  const updateChatData = (updatedMessage) => {
+    const messages = chatMessagesData.map((message) => {
+      if (message.id === updatedMessage.id) {
+        return updatedMessage;
+      } else {
+        return message;
+      }
+    });
+
+    setChatMessagesData(messages);
+  };
```

Comment:
We use this method to update how a chat entry renders after someone likes an entry. To constrain the scope of this method so that it is only responsible for re-rendering a chat entry after it's been liked, we might refactor this method so that we're not returning the entire message. 

updateChatData could become something like: 
``` javascript
const updateHeart = (id) => {
    const chatMessagesCopy = [...chatMessagesData];
    for (let entry of chatMessagesCopy) {
      if (entry.id === id) {
        entry.heart = !entry.heart; 
      }
    }
    setChatMessagesData(chatMessagesCopy);
  };
```

If you do this, then you can update ChatEntry.js too so you only need to pass the prop `id` instead of the whole message object